src/components/withdrawals/CreateForm.tsx

```diff
@@ -0,0 +1,219 @@
+  const router = useRouter()
+  const bankAccounts = useBankAccountsForWithdrawal()
+  const vaults = useVaultsForWithdrawal()
+  const campaigns = useCampaignsForWithdrawal()
+  const personList = usePersonListForWithdrawal()
+  const currencies = ['BGN', 'USD', 'EUR']
+  const [currency, setCurrency] = useState('')
+  const [amount, setAmount] = useState(Number)
+  const [reason, setReason] = useState('')
+  const [approvedById, setApprovedById] = useState('')
+  const [bankAccountId, setBankAccountId] = useState('')
+  const [vaultId, setVaultId] = useState('')
+  const [campaignId, setCampaignId] = useState('')
+
+  const { t } = useTranslation()
+
+  const mutationFn = useCreateWithdrawal()
+
+  const mutation = useMutation<
+    AxiosResponse<WithdrawalResponse>,
+    AxiosError<ApiErrors>,
+    WithdrawalInput
+  >({
+    mutationFn,
+    onError: () => AlertStore.show(t('withdrawals:alerts:error'), 'error'),
+    onSuccess: () => {
+      AlertStore.show(t('withdrawals:alerts:create'), 'success')
+      router.push(routes.admin.withdrawals.index)
+    },
+  })
+
+  function handleSubmit() {
+    const data: WithdrawalInput = {
+      status: WithdrawalStatus.initial,
+      currency: currency,
+      amount: amount,
+      reason: reason,
+      sourceVaultId: vaultId,
+      sourceCampaignId: campaignId,
+      bankAccountId: bankAccountId,
+      documentId: '6061b425-34e6-45bf-b563-a85b933c9339',
+      approvedById: approvedById,
+    }
+
+    if (
+      (data.reason != undefined && data.reason.length < 4) ||
+      (data.reason != undefined && data.reason?.length > 50)
+    ) {
+      AlertStore.show('Причината трябва да бъде между 4 и 50 символа!', 'error')
+    } else if (data.sourceVaultId != undefined && data.sourceVaultId?.length < 1) {
+      AlertStore.show('Моля изберете Трезор!', 'error')
+    } else if (data.sourceCampaignId != undefined && data.sourceCampaignId?.length < 1) {
+      AlertStore.show('Моля изберете Кампания!', 'error')
+    } else if (data.bankAccountId != undefined && data.bankAccountId?.length < 1) {
+      AlertStore.show('Моля изберете Банков акаунт!', 'error')
+    } else if (data.currency != undefined && data.currency?.length < 1) {
+      AlertStore.show('Моля изберете Валута!', 'error')
+    } else if (data.amount != undefined && data.amount < 1) {
+      AlertStore.show('Моля изберете Сума!', 'error')
+    } else if (data.documentId != undefined && data.documentId?.length < 1) {
+      AlertStore.show('Моля изберете Документ!', 'error')
+    } else if (data.approvedById != undefined && data.approvedById?.length < 1) {
+      AlertStore.show('Моля изберете полето "Одобрен от"!', 'error')
+    } else {
+      mutation.mutate(data)
+    }
+  }
+
+  return (
+    <Box>
+      <Typography variant="h5" component="h2" sx={{ marginBottom: 2, textAlign: 'center' }}>
+        {t('withdrawals:form-heading')}
+      </Typography>
+      <Grid container spacing={2} sx={{ width: 600, margin: '0 auto' }}>
```

Comment:
We better use `GenericForm` to handle the form data and the validation

Handling it manually will lead to confusion and duplicate data.

The Formik forms can handle validation and can set values on their own without custom `useState` for each of the data fields